### PR TITLE
[HOLD] HELPS-3395: Loop Config - add into webpack

### PR DESCRIPTION
**Pull request by JosefBud-olive**
### Changes made:

- Added support for new `configSchema` property in a Loop's package.json's `ldk` object
- Added support for future Config Aptitude permissions
- Validating `configSchema` matches [JSON Schema](https://json-schema.org/) standard using [ajv](https://github.com/ajv-validator/ajv)
- Updated self test Loop's package.json with testable `configSchema` for when we add a Config Aptitude
- Added a PR template

### CODE REVIEW

File: ldk/javascript/src/webpack/generate-banner.test.ts [hunk 1]
```diff
@@ -45,9 +75,39 @@ describe('Generate Banner', () => {
     const actual = getLoopMetadataContent(generateBanner(ldkSettings));
     const expected = {
       oliveHelpsContractVersion: '0.3.0',
+      configSchema: {
+        $schema: 'http://json-schema.org/draft-07/schema#',
```
Comment (alexmatthew-olive) [inline ldk/javascript/src/webpack/generate-banner.test.ts:79]: This is fine that is here but we shouldn't require this property, I assume leaving this off still works fine correct?

Comment (JosefBud-olive) [reply ldk/javascript/src/webpack/generate-banner.test.ts:79]: Yep! Everything will go through fine without that, it's not required

**Review by MaggieMaSS — APPROVED** [review]

File: ldk/javascript/src/webpack/generate-banner.ts [hunk 2]
```diff
@@ -10,11 +11,47 @@ export function generateMetadata(ldkSettings: LdkSettings): string {
...
+    const errors: string[] = [];
+
+    Object.values(ldkSettings.ldk.configSchema.properties as LdkConfigProperties).forEach(
+      (configChild) => {
+        try {
+          ajv.compile(configChild);
+
+          const { default: defaultValue, type } = configChild;
+
+          // Only allow types supported by Loop Library form
```
Comment (mimoduo) [inline ldk/javascript/src/webpack/generate-banner.ts:26]: Which loop library form?

File: ldk/javascript/src/webpack/generate-banner.ts [hunk 3]
```diff
@@ -10,11 +11,47 @@ export function generateMetadata(ldkSettings: LdkSettings): string {
...
+          if (type !== 'string' && type !== 'object') {
+            errors.push(
+              `Error: The LDK does not currently support ${type} types in the config schema`,
+            );
+          }
+
+          // Check to make sure default value matches type because ajv won't
+          if (defaultValue && typeof defaultValue !== type) {
+            errors.push(`Error: The default value ${defaultValue} does not match the type ${type}`);
```
Comment (mimoduo) [inline ldk/javascript/src/webpack/generate-banner.ts:35]: Would be a little easier to identify the `defaultValue` if it had quotes or something!

File: ldk/javascript/src/webpack/generate-banner.ts (hunk 2 above)
Comment (JosefBud-olive) [reply ldk/javascript/src/webpack/generate-banner.ts:26]: Heya, Bryan! The form is mentioned [here](https://coda.io/d/Loop-Config_dE8VVdCUd8q/Loop-Config_suO_h#_luGNs) but @alexmatthew-olive would have more detail. It's the form that will be generated from the schema, that org admins will fill out to define the variables.

File: ldk/javascript/src/webpack/generate-banner.ts [hunk 4]
```diff
@@ -10,11 +11,49 @@ export function generateMetadata(ldkSettings: LdkSettings): string {
...
+        } catch (error) {
+          errors.push(error as string);
+        }
+      },
+    );
+
+    if (errors.length) {
+      throw new Error(`There was an error with your LDK config:\n${errors.join('\n')}`);
+    }
+  }
```
Comment (ulricesparra-oliveai) [inline ldk/javascript/src/webpack/generate-banner.ts:48]: Turns out we're simplifying your job and supporting any valid JSON schema in the form. Should only need to grab the configSchema value, try compiling it with ajv, and throw an error if that fails
```suggestion
  if (ldkSettings.ldk.configSchema) {
    const ajv = new Ajv();

    try {
      ajv.compile(ldkSettings.ldk.configSchema);
    } catch (error) {
      throw new Error(`There was an error with your LDK config:\n${error}`);
    }
  }
```

Comment (JosefBud-olive) [reply ldk/javascript/src/webpack/generate-banner.ts]: Sounds good! Changed this in https://github.com/open-olive/loop-development-kit/pull/519/commits/a9766d413c596be8470a4cc221532c3bfd940519

**Review by alexmatthew-olive — APPROVED** [review]

**Pull request merged**